### PR TITLE
Update notebooks

### PR DESCRIPTION
**Pull request by bramalingam**
Requires https://github.com/IDR/idr-py/pull/8 and https://github.com/IDR/jupyter-docker/pull/22
and https://github.com/ome/rOMERO-gateway/pull/36 for testing.
Will push a docker image which combines all 3 PR's for testing purposes.

Fixes all notebooks which have import issues or deprecated methods (from outdated dependency versions)

Going to add versions for all python libraries in the docker file to make the Jupyter installation robust and reproducible (for testing/using the notebooks).

@joshmoore @jburel @manics

### CODE REVIEW

**Comment by jburel** [comment]
This does not seem to be included in the docker image https://hub.docker.com/r/bramalingam/idr_jupyter/
I still have the figure notebook

**Comment by bramalingam** [comment]
Apologies, the open PR's were not included: have updated the docker file (Temporarily) to consume the open PR's from my branches.
https://github.com/IDR/jupyter-docker/pull/22/commits/6f80c39f6b143a6be141950cc3d1d1f7561f62a2

And have also pushed a new image to,
https://hub.docker.com/r/bramalingam/idr_jupyter/tags/
"idr_jupyter_v03"

**Comment by jburel** [comment]
Condensation BulkAnnotations
The notebook points at ``idr-testing`` this will need to be changed.

* This notebooks shows... Bullet point should start with Capital
* rOMERO-gateway
* [idr0002-heriche-condensation/screenA](http://10.0.51.164/webclient/?show=screen-102) link is not working
* The article ...
* [CDC20](http://10.0.51.164/webclient/img_detail/250083/) link not working
* ``More than two Short or Long phenotype replicates --> `` replaces ``-->`` by ``implies that the gene is involved...``
* Is 5.3.3 in used or 5.3.5?
* Step 1, nicer to remove warning message
* ``E. g. load data for plate 1`` extra space ``E.g.`` 
* ``e. g. how many controls`` same
* Last step before Replicate some of the articles figures ```Warning message:
“Removed 2 rows containing missing values (position_stack).”```

File: notebooks/CondensationBulkAnnotations.R.ipynb [hunk 1]
```diff
@@ -79,14 +94,14 @@
...
-   "execution_count": null,
+   "execution_count": 2,
    "metadata": {
     "scrolled": true
    },
    "outputs": [],
    "source": [
-    "source('../library/idr.R')\n",
-    "ome <- connection()"
+    "server <- OMEROServer(host = 'idr-testing.openmicroscopy.org', username = Sys.getenv('IDR_USER'), password = Sys.getenv('IDR_PASSWORD'), port = as.integer(4064))\n",
```
Comment (jburel) [inline notebooks/CondensationBulkAnnotations.R.ipynb]: this needs to be changed to prod server

File: notebooks/CondensationBulkAnnotations.R.ipynb [hunk 2]
```diff
@@ -191,11 +232,96 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 6,
    "metadata": {
     "scrolled": false
    },
-   "outputs": [],
+   "outputs": [
```
Comment (jburel) [inline notebooks/CondensationBulkAnnotations.R.ipynb]: why do you include the outputs?

File: notebooks/CondensationBulkAnnotations.R.ipynb [hunk 3]
```diff
@@ -216,11 +342,30 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 7,
    "metadata": {
     "scrolled": true
    },
-   "outputs": [],
+   "outputs": [
```
Comment (jburel) [inline notebooks/CondensationBulkAnnotations.R.ipynb]: same

File: notebooks/CondensationBulkAnnotations.R.ipynb [hunk 4]
```diff
@@ -277,11 +422,347 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 9,
    "metadata": {
     "scrolled": false
    },
-   "outputs": [],
+   "outputs": [
```
Comment (jburel) [inline notebooks/CondensationBulkAnnotations.R.ipynb]: same

**Comment by jburel** [comment]
I do not think it is a good to include rows and rows of output in the notebook but it is the approach taken the long table should be formatted better e.g. http://localhost:8888/notebooks/notebooks/CalculateSharpness.ipynb for putting the results in a scrollpane

File: notebooks/CondensationBulkAnnotations.R.ipynb [hunk 5]
```diff
@@ -61,11 +61,26 @@
...
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Loading required package: rJava\n",
+      "Loading required package: jpeg\n",
+      "Warning message:\n",
+      "“replacing previous import ‘rJava::tail’ by ‘utils::tail’ when loading ‘romero.gateway’”Warning message:\n",
```
Comment (dominikl) [inline notebooks/CondensationBulkAnnotations.R.ipynb]: These warning should be gone now with https://github.com/ome/rOMERO-gateway/pull/34 being merged.

**Comment by jburel** [comment]
Figure_1_Sampling_of_Phenotypes.ipynb
* ``This notebook exemplifies how to reproduce Figure 1 of the article.`` Which article? Add link to another notebook and articke
*  Having message ``Connected to IDR... `` without client on connection step is confusing
* Build and display figure step is painfully slow making this notebook not really usable

**Comment by jburel** [comment]
GeneNetwork.ipynb

A bit of background will be nice, since it assumes that the user is familiar with the previous notebooks
* networkx (does not work)
```
/opt/conda/envs/python2/lib/python2.7/site-packages/networkx/drawing/nx_pylab.py:126: MatplotlibDeprecationWarning: pyplot.hold is deprecated.
    Future behavior will be consistent with the long-time default:
    plot commands add elements without first clearing the
    Axes and/or Figure.
  b = plt.ishold()
```

**Comment by sbesson** [comment]
Discussed https://github.com/IDR/idr-notebooks/pull/47#issuecomment-361210301 with various members of team. In the scope of 0.4.5, the decisions was to keep the output of the cells so that they are rendered in GitHub but keep them reasonably sized i.e. displaying only the  N first  rows. As a rule of thumb, each cell output should fit within size of a browser window. 

Future steps might include clearing the output in the VAE environments but this might require additional step or separate branches.

File: notebooks/CondensationBulkAnnotations.R.ipynb [hunk 6]
```diff
@@ -100,7 +100,7 @@
    },
    "outputs": [],
    "source": [
-    "server <- OMEROServer(host = 'idr-testing.openmicroscopy.org', username = Sys.getenv('IDR_USER'), password = Sys.getenv('IDR_PASSWORD'), port = as.integer(4064))\n",
+    "server <- OMEROServer(host = Sys.getenv('IDR_HOST'), username = Sys.getenv('IDR_USER'), password = Sys.getenv('IDR_PASSWORD'), port = as.integer(4064))\n",
```
Comment (jburel) [inline notebooks/CondensationBulkAnnotations.R.ipynb]: The port could be different too
This should be retrieved via system property and if not set fall back to the default port

Comment (manics) [reply notebooks/CondensationBulkAnnotations.R.ipynb]: Use `IDR_PORT` if it's set https://github.com/IDR/idr-py/blob/v0.1.1/idr/connections.py#L43

Comment (manics) [reply notebooks/CondensationBulkAnnotations.R.ipynb]: Though we may want to switch to `IDR_OMERO_CONFIGURATION_URL` https://github.com/IDR/idr-py/blob/v0.1.1/idr/connections.py#L49

**Comment by sbesson** [comment]
Review all notebooks modified by this PR comparing them to the current notebooks of http://idr.openmicroscopy.org/jupyter/. At the moment, the notebooks are  on IDR 0.4.4 incl. CondensationBulkAnnotations (missing kernel),  Fig1_Sampling (import error on first cell), GeneNetwork (error during the Gene query cell),

Overall these changes combine with https://github.com/IDR/jupyter-docker/pull/22 represent an improvement. A few general comments:
- the reduction of some long cell output to the  first N rows only helps
- the systematic usage of the idr.py widgets is an improvement in terms of UX for known long-running cells

Below are a summary of changes by playbook with an assessment of the severity. For IDR 0.4.5, I would focus on blocker and probably high first. Others can be dealt with if there is time or captured elsewhere.

| Notebook                    | Severity | Description                                                                                                 |
|-----------------------------|----------|-------------------------------------------------------------------------------------------------------------|
| CalculateSharpness          | high     | Hardcoded tadpole.openmciroscopy.org server                                                                 |
| CalculateSharpness          | medium   | In the writing section, could use a sentence explaining this only works for OMERO 5.3.         |
| CondensationBulkAnnotations | low      | Red messages slightly unfriendly                                                                                                |
| CondensationBulkAnnotations | blocker  | Quality control and well position cell crashes with `Error in eval(expr, envir, enclos): object 'chisq_test_frame' not found` |
| CondensationBulkAnnotations | low      |  Section about figure could use direct link to http://www.molbiolcell.org/content/25/16/2522.full#F3                                   |
| CondensationBulkAnnotations | low      | Some long running cells (thumbnail plotting, data loading)                                                                                    |
| Fig1_Sampling               | low      | The "Build and display figure" heading reads as it applies   to the following cell. Slightly confusing                                            |
| GeneNetwork                 | medium   | Could use reference and/or hyperlink to the figure                                         |
| QueryIDRWithGeneLists       | medium   | UX: the way to specify the GO ontology is confusing (editing output rather than the cell) |
| QueryIDRWithGeneLists       | medium   | Slightly confused by the absence of GO ontology default while we  explicitly provide one for the taxonomy     |
| QueryIDRWithGeneLists       | high     | Only display the first 10 genes for long lists                                                           |
| QueryIDRWithGeneLists       | blocker  | Cell crashes with `phenMap is not defined`                                                                  |
| All       | high  | Extend copyright range to 2018 |                                                                  |

Given the size of this PR, I would suggest we get this merged as phase 1 and start going through the critical issues in a follow-up PRs (ideally per notebook?)

**Pull request merged**